sdk/core/azure-core-amqp/src/amqp/connection.cpp

```diff
@@ -98,6 +98,59 @@ namespace Azure { namespace Core { namespace Amqp { namespace _internal {
   {
     m_impl->SetIdleEmptyFrameSendPercentage(ratio);
   }
+  std::ostream& operator<<(std::ostream& stream, ConnectionState state)
+  {
+    switch (state)
```

Comment:
Optional: you can write this using a map
```cpp
namespace
{
    std::map<ConnectionState, std::string> const ConnectionStateToString{ {ConnectionState::Start, "Start"}, ... };
}

std::ostream& operator<<(std::ostream& stream, ConnectionState state)
{
    auto const toString = ConnectionStateToString.find(state);
    if (toString != ConnectionStateToString.end())
    {
        stream << toString.second;
    }

    return stream;
}
```

Reply:
I thought about doing that, but writing it as a map means that we burn CPU cycles at startup initializing the static. I'd rather have a big switch statement where the cost is burned only when someone tries to insert a `ConnectionState` in an `ostream` than have a static map which is initialized whenever anyone uses the package.

We could lazy populate the map, but that's just almost as bad as having the switch statement.

Reply:
Oh, and it introduces concurrency issues that would have to be addressed.

Reply:
Hmm, the map would be const, so hopefully no concurrency issues, but I see how it is potentially more problematic.